kaiax/builder/impl/init.go

```diff
@@ -35,17 +35,35 @@ type InitOpts struct {
 	Backend api.Backend
 }
 
-type knownTxs map[common.Hash]knownTx
+type knownTxs map[common.Hash]*knownTx
 
-func (k knownTxs) add(tx *types.Transaction) {
-	k[tx.Hash()] = knownTx{
-		tx:        tx,
-		time:      time.Now(),
-		isDemoted: false,
+func (k knownTxs) add(tx *types.Transaction, status int) {
+	if tx == nil {
+		return
 	}
+
+	if ktx, ok := k[tx.Hash()]; ok {
+		ktx.status = status
```

Comment:
The time doesn't change if already existing tx comes in. Is this intended change?